spyder/plugins/updatemanager/workers.py

```diff
@@ -445,7 +480,150 @@ def start(self):
             # running).
             try:
-                self.sig_ready.emit()
+                self.sig_ready.emit(self.error is None)
+            except RuntimeError:
+                pass
+
+
+class WorkerUpdateUpdater(BaseWorker):
+    """
+    Worker that checks and updates Spyder-updater without blocking
+    the Spyder user interface.
+    """
+
+    def __init__(self, stable_only):
+        super().__init__()
+        self.stable_only = stable_only
+        self.asset_info = None
+        self.installer_path = None
+        self.error = None
+
+    def _check_asset_available(self):
+        """Checks if there is an update available for the Updater."""
+
+        # Get release info from Github
+        releases = get_github_releases(updater=True)
+
+        if self.stable_only:
+            # Only use stable releases
+            releases = {
+                k: v for k, v in releases.items() if not k.is_prerelease
+            }
+        logger.debug(f"Available releases: {sorted(releases)}")
+
+        self.asset_info = _check_asset_available(releases, updater=True)
+
+    def _clean_installer_dir(self):
+        """Remove downloaded file"""
+        installer_dir = osp.dirname(self.installer_path)
+        if osp.exists(installer_dir):
+            try:
+                shutil.rmtree(installer_dir)
+            except OSError as err:
+                logger.debug(err, stack_info=True)
+
+    def _download_asset(self):
+        """Download Updater lock file"""
+        self.installer_path = osp.join(
+            get_temp_dir(),
+            "updates",
+            "spyder-updater",
+            str(self.asset_info["version"]),
+            self.asset_info["filename"],
+        )
+
+        if (
+            osp.exists(self.installer_path)
+            and validate_download(
+                self.installer_path, self.asset_info["checksum"]
+            )
+        ):
+            logger.debug(f"{self.installer_path} already downloaded.")
+            return
+
+        self._clean_installer_dir()
+        dirname = osp.dirname(self.installer_path)
+        os.makedirs(dirname, exist_ok=True)
+
+        url = self.asset_info["url"]
+        logger.info(f"Downloading {url} to {self.installer_path}")
+
+        page = requests.get(url, headers=GH_HEADERS)
+        page.raise_for_status()
+        with open(self.installer_path, 'wb') as f:
+            f.write(page.content)
+
+        if validate_download(self.installer_path, self.asset_info["checksum"]):
+            logger.info('Download successfully completed.')
+            with ZipFile(self.installer_path, 'r') as f:
+                f.extractall(dirname)
+        else:
+            raise UpdateDownloadError("Download failed!")
+
+    def _install_update(self):
+        """Install or update Spyder-updater environment."""
+        dirname = osp.dirname(self.installer_path)
+
+        if os.name == "nt":
+            plat = "win-64"
+        if sys.platform == "darwin":
```

Comment:
I think the error I experienced is caused due to this being an `if`. Probably changing it to `elif` should be enough?

```suggestion
        elif sys.platform == "darwin":
```

Reply:
Oops! Thanks!